workflow/controller/operator.go

```diff
@@ -263,6 +264,8 @@ func (woc *wfOperationCtx) operate(ctx context.Context) {
 
 	if woc.wf.Status.Phase == wfv1.WorkflowUnknown {
 		woc.markWorkflowRunning(ctx)
+		setWfPodNamesAnnotation(woc.wf)
```

Comment:
apparently there's a small bug here since line 238 above wasn't updated with this. See https://github.com/argoproj/argo-workflows/issues/10178#issuecomment-2068596407

Working on a fix for that

Reply:
Fixed in #12965